### PR TITLE
Change dependency on avro-to-typescript to forked version. Fix tests.

### PR DESCRIPTION
**Pull request by chasdevs**
This PR changes @degordian/avro-to-typescript to my forked @chasdevs/avro-to-typescript (which does not have the schema registry and node-rdkafka dependencies). I also cleaned up some tests along the way to get circleci passing and executing faster.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ jobs:
                   KAFKA_ZOOKEEPER_CONNECT: 'localhost:2181'
                   KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT
                   KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://localhost:9092,PLAINTEXT_HOST://localhost:29092
+                  KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
```
Comment (chasdevs) [inline .circleci/config.yml]: Made tests faster and more reliable (something something consumer groups stackoverflow)

File: src/consumer/Consumer.ts [hunk 2]
```diff
@@ -61,4 +60,9 @@ export default class Consumer extends BaseKafkaClient {
         const handledRecordTypes = handlers.map(handler => handler.getMessageType().name);
         return [...new Set(handledRecordTypes)].map(name => TopicFormatter.topicFromRecordName(name));
     };
+
+    public async disconnect(): Promise<void> {
+        await super.disconnect();
+        await this.recordProcessor.getFailureProducer().disconnect();
+    }
```
Comment (chasdevs) [inline src/consumer/Consumer.ts]: Disconnect the consumer along with its failureProducer.

File: src/test/common/endToEnd/endToEnd.test.ts [hunk 3]
```diff
@@ -5,96 +5,80 @@ import ConsumerBuilder from '../../../consumer/ConsumerBuilder';
...
+        const consumer = new ConsumerBuilder(brokers, groupId, schemaRegistryUrl).build();
 
-        const fakeRecord = FakeFactory.fakeRecord();
-        await producer.produce(fakeRecord, topic);
-        await producer.disconnect();
+        await producer.produce(producedRecord, topic);
 
-        const builder = new ConsumerBuilder(brokers, groupId, schemaRegistryUrl);
-        const consumer = builder.build();
+        await new Promise(resolve => {
```
Comment (chasdevs) [inline src/test/common/endToEnd/endToEnd.test.ts]: Using promises instead of intervals.

File: src/test/producer/integration/Producer.test.ts [hunk 4]
```diff
@@ -1,32 +1,34 @@
...
 describe('Producer integration tests', () => {
+    let producer: Producer;
     const brokers = config.get('test.brokerUrls');
     const schemaRegistryUrl = config.get('test.schemaRegistryUrl');
 
-    beforeAll(() => jest.setTimeout(60 * 1000));
-
     test('Error is thrown if kafka client is unable to connect to broker', async () => {
-        const producer = new ProducerBuilder(['1.2.3.4'], schemaRegistryUrl).build();
+        producer = new ProducerBuilder(['1.2.3.4'], schemaRegistryUrl).setConnectTimeout(10).build();
```
Comment (chasdevs) [inline src/test/producer/integration/Producer.test.ts]: Set the timeout to 10ms so this test takes less than 30s

File: .circleci/config.yml (hunk 1 above)
Comment (balexander) [reply .circleci/config.yml]: 👍

File: src/consumer/Consumer.ts (hunk 2 above)
Comment (balexander) [reply src/consumer/Consumer.ts]: 👍

File: src/test/common/endToEnd/endToEnd.test.ts (hunk 3 above)
Comment (balexander) [reply src/test/common/endToEnd/endToEnd.test.ts]: 👍 That was on my todo list at one point but I forgot about it -- thanks for the cleanup

**Pull request merged**